fenixedu-ist-giaf-invoices/src/main/java/pt/ist/fenixedu/domain/documents/LiquidationLetter.java

```diff
@@ -0,0 +1,113 @@
+
+        json.addProperty("limitDate",  dueDate.toString(DATE_FORMAT));
+
+        final Money debtTotal = reportEntry.amount.add(reportEntry.interest);
+        json.addProperty("debtValue", reportEntry.amount.toPlainString());
+        json.addProperty("debtInterest", reportEntry.interest.toPlainString());
+        json.addProperty("debtTotal", debtTotal.toPlainString());
+        json.addProperty("debtInitialDate", reportEntry.dueDate.toString(DATE_FORMAT));
+
+        json.addProperty("presidentGender", "male");
+
+        final EventPaymentCodeEntry eventPaymentCodeEntry = EventPaymentCodeEntry.create(event, debtTotal);
+        eventPaymentCodeEntry.setDueDate(dueDate.plusDays(7));
+        final PaymentCode paymentCode = eventPaymentCodeEntry.getPaymentCode();
+        json.addProperty("mbEntity", paymentCode.getEntityCode());
+        json.addProperty("mbRef", paymentCode.getFormattedCode());
+
+
+        return json;
+    }
+
+    private static String notificationRefernece(final Event event, final LocalDate notificationDate) {
+        return Integer.toString(notificationDate.getYear()) + "DA/PP/" + event.getExternalId();
+    }
+
+    static void addPersonalInformation(final JsonObject json, final Event event, final Person person) {
+        final String clientTIN = ClientMap.uVATNumberFor(person);
+        final String[] address = toAddress(clientTIN, person);
+
+        json.addProperty("gender", person.isMale() ? "male" : "female");
+        json.addProperty("name", person.getName());
+        json.addProperty("addressLine1", address[0]);
+        json.addProperty("addressLine2", address[1]);
+        json.addProperty("addressLine3", address[2]);
+        json.addProperty("tin", clientTIN);
+        json.addProperty("username", person.getUsername());
+    }
+
+    private static String[] toAddress(final String clientTIN, final Person person) {
+        final String countryCode = clientTIN.substring(0, 2);
+        final PhysicalAddress physicalAddress = Utils.toAddress(person, countryCode);
+        final String street = physicalAddress != null && physicalAddress.getAddress() != null
+                && !Strings.isNullOrEmpty(physicalAddress.getAddress().trim()) ?
+                    Utils.limitFormat(MAX_SIZE_ADDRESS, physicalAddress.getAddress()) : UNKNOWN;
+
+        final String city = Utils.limitFormat(MAX_SIZE_CITY, person.getDistrictSubdivisionOfResidence()).trim();
+        final String cityString = !Strings.isNullOrEmpty(city) ? city : UNKNOWN;
+        final String region = Utils.limitFormat(MAX_SIZE_REGION, person.getDistrictOfResidence()).trim();
+        final String regionString = !Strings.isNullOrEmpty(region) ? region : UNKNOWN;
+        String postalCode =
+                physicalAddress == null ? null : Utils.limitFormat(MAX_SIZE_POSTAL_CODE, physicalAddress.getAreaCode()).trim();
+        //sometimes the address is correct but the vatNumber doesn't exists and a random one was generated from the birth country
```

Comment:
this way of operating regarding the postal code is specific for SAP, I believe that in this case we want the true postal code of the address, otherwise the address may end up with an address of a country and a postal code of another and not receiving the letter.